src/metabase/lib/schema/aggregation.cljc

```diff
@@ -5,29 +5,71 @@
    [metabase.util.malli.registry :as mr]))
 
 ;; count has an optional expression arg
-(mbql-clause/define-catn-mbql-clause :count
-  [:expression [:? [:ref ::expression/number]]])
+(mbql-clause/define-catn-mbql-clause :count :- :type/Integer
+  [:expression [:? [:schema [:ref ::expression/number]]]])
 
-(defmethod expression/type-of* :count
-  [[_tag _opts expr]]
-  (if-not expr
-    :type/Integer
-    (expression/type-of expr)))
+(mbql-clause/define-tuple-mbql-clause :avg :- :type/Float
+  [:schema [:ref ::expression/number]])
+
+(mbql-clause/define-tuple-mbql-clause :distinct :- :type/Integer
+  [:schema [:ref ::expression/expression]])
+
+(mbql-clause/define-tuple-mbql-clause :count-where :- :type/Integer
+  [:schema [:ref ::expression/boolean]])
+
+(mbql-clause/define-tuple-mbql-clause :max
+  [:schema [:ref ::expression/number]])
+
+(expression/register-type-of-first-arg :max)
+
+(mbql-clause/define-tuple-mbql-clause :median
+  [:schema [:ref ::expression/number]])
+
+(expression/register-type-of-first-arg :median)
+
+(mbql-clause/define-tuple-mbql-clause :min
+  [:schema [:ref ::expression/number]])
+
+(expression/register-type-of-first-arg :min)
+
+(mbql-clause/define-tuple-mbql-clause :percentile
+  #_expr [:schema [:ref ::expression/number]]
+  #_percentile [:schema [:ref ::expression/non-integer-real]])
```

Comment:
Cannot we allow integers like 90?

Reply:
Percentile should be in range (0.0, 1.0).